logstash-core/src/main/java/org/logstash/common/io/DeadLetterQueueReader.java

```diff
@@ -65,6 +65,9 @@ public void seekToNextEvent(Timestamp timestamp) throws IOException {
             currentReader = new RecordIOReader(segment);
             byte[] event = currentReader.seekToNextEventPosition(timestamp, (b) -> {
                 try {
+                    if (b == null){
```

Comment:
Yeah, I think you're right - this was from an early iteration where I didn't handle the null in RecordIOReader. I can remove it